source/Octopus.Tentacle/Maintenance/WorkspaceCleaner.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Communications;
+using Octopus.Tentacle.Scripts;
+using Octopus.Tentacle.Services.Scripts;
+using Octopus.Time;
+
+namespace Octopus.Tentacle.Maintenance
+{
+    public class WorkspaceCleaner
+    {
+        readonly WorkspaceCleanerConfiguration configuration;
+        readonly IScriptWorkspaceFactory scriptWorkspaceFactory;
+        readonly IClock clock;
+        readonly ISystemLog log;
+
+        readonly ScriptService scriptService;
+        readonly ScriptServiceV2 scriptServiceV2;
+
+        public WorkspaceCleaner(
+            WorkspaceCleanerConfiguration configuration,
+            IScriptWorkspaceFactory scriptWorkspaceFactory,
+            IServiceRegistration serviceRegistration,
+            IClock clock,
+            ISystemLog log)
+        {
+            this.configuration = configuration;
+            this.scriptWorkspaceFactory = scriptWorkspaceFactory;
+            this.clock = clock;
+            this.log = log;
+
+            scriptService = serviceRegistration.GetService<ScriptService>();
+            scriptServiceV2 = serviceRegistration.GetService<ScriptServiceV2>();
+        }
+
+        public async Task Clean(CancellationToken cancellationToken)
+        {
+            var deleteWorkspacesOlderThanDateTimeUtc = clock.GetUtcTime().DateTime.Subtract(configuration.DeleteWorkspacesOlderThanTimeSpan);
+            log.Info($"Cleaning orphaned workspaces older than {deleteWorkspacesOlderThanDateTimeUtc:g}");
+
+            var deletedCount = 0;
+            var workspaces = scriptWorkspaceFactory.GetUncompletedWorkspaces();
+            foreach (var workspace in workspaces)
+            {
+                try
+                {
+                    if (scriptService.IsRunningScript(workspace.ScriptTicket)) continue;
+                    if (scriptServiceV2.IsRunningScript(workspace.ScriptTicket)) continue;
+
+                    var workspaceLogFilePath = workspace.LogFilePath;
+                    if (!File.Exists(workspaceLogFilePath)) continue;
+
+                    var outputLogFileLastWriteTimeUtc = File.GetLastWriteTimeUtc(workspaceLogFilePath);
```

Comment:
Is there a race condition here that we could handle gracefully rather than log error when the file is deleted after we check it exists but before we read it's properties?

Thought process is to avoid Errors in the logs that might concern users when the cause is just a race condition.

Reply:
Sounds good. Done

Reply:
@nathanwoctopusdeploy , after doing some manual testing, I came to realize that [File.GetLastWriteTimeUtc](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.getlastwritetimeutc?view=net-7.0) doesn't actually throw an exception when the file doesn't exist. It returns a date from 1601.

So the try/catch I had around here was redundant. 

I think instead, we simply check to see if the file still exists after we call `File.GetLastWriteTimeUtc` (because if it's not there, then it shows the race condition is happening).

Let me know if you think we should put the try/catch back, but I think this is a better solution.